### PR TITLE
work in progress: partially working invitation function

### PR DESCRIPTION
**Pull request by kay-lee**
- yet to have functionality to render previously invited friends
- name/email submission and db connection is functional

### CODE REVIEW

File: server/routes/invitee_list.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+let db = require('../config');
+
+module.exports = function(req, res) {
+  db.query('SELECT * FROM invites')
```
Comment (misteryeo) [inline server/routes/invitee_list.js]: Should eventually have this select only invites which match the user_id of user who is logged in

File: server/schema.js [hunk 2]
```diff
@@ -46,5 +39,10 @@ module.exports = (db) => {
     return db.query('CREATE TABLE IF NOT EXISTS friends (\
       user1 BIGINT not null,\
       user2 BIGINT not null);')
+  })
+  .then(() => {
+    return db.query('CREATE TABLE IF NOT EXISTS invites (\
```
Comment (misteryeo) [inline server/schema.js]: Would suggest having an auto incrementing primary key value for invites to be used in a potential join table later.

File: src/client/app/Invite.jsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+			email: this.state.email
+		}
+		console.log('invitee', invitee);
+		$.ajax({
+			url: '/invites',
+			method: 'POST',
+			data: JSON.stringify(invitee),
+			contentType: 'application/json',
+			success: function(data) {
+				this.hideModal();
```
Comment (misteryeo) [inline src/client/app/Invite.jsx]: Wondering if the modal should be hidden each time user submits. What if they'd like to add another friend? They'd have to repeat the process from the beginning each time.

File: server/routes/invitee_list.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  .then ( (users) => {
+    res.status(201).json(users);
+  })
+  .catch( (err) => {
+    res.status(400).send(err, 'Error in get_users handler function');
```
Comment (iccir919) [inline server/routes/invitee_list.js]: error in get_invites

File: src/client/app/InviteeList.jsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import {render} from 'react-dom';
+import Modal from 'boron/DropModal';
+import $ from 'jquery';
+import InviteeListEntry from './InviteeListEntry.jsx';
+
+class InviteeList extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+
```
Comment (misteryeo) [inline src/client/app/InviteeList.jsx]: Not sure if you have any state to add here?

File: src/client/app/InviteeList.jsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+	}
+
+	getInviteeList () {
+		var _this = this;
+		$.ajax({
+			url: '/invites',
+			method: 'GET',
+			contentType: 'application/json',
+			success: function(data) {
+				console.log(data);
```
Comment (misteryeo) [inline src/client/app/InviteeList.jsx]: Where does the data go upon success?

File: src/client/app/InviteeListEntry.jsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+		this.state = {
+		}
+		this.getInviteeList = this.getInviteeList.bind(this);
+	}
+
+	componentDidMount () {
+		this.getInviteeList();
+	}
+
+	getInviteeList () {
```
Comment (misteryeo) [inline src/client/app/InviteeListEntry.jsx]: Is this a duplicate method in a child component?

File: src/client/styles.css [hunk 8]
```diff
@@ -186,6 +186,18 @@ button */
   border-radius: 12px;
 }
 
+.invite_friends {
+  font-family: sans-serif;
+  font-size: 32px;
+  color: white;
+  background-color: #f47676;
```
Comment (iccir919) [inline src/client/styles.css]: i like your color choice

File: server/routes/invites.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
+'use strict'
+
+let db = require('../config.js');
+
+module.exports = function(req, res) {
+  db.query(`INSERT into INVITES (name, email) VALUES ('${req.body.name}', '${req.body.email}')`)
+  .then( (result) => {
```
Comment (iccir919) [inline server/routes/invites.js]: how are you linking the user of the invite to the information of the invite?

File: src/client/app/InviteeList.jsx (hunk 6 above)
Comment (kay-lee) [inline src/client/app/InviteeList.jsx]: Nowhere yet :O I forgot to update state with the data.

File: src/client/app/InviteeListEntry.jsx [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+		var _this = this;
+		$.ajax({
+			url: '/invites',
+			method: 'GET',
+			contentType: 'application/json',
+			success: function(data) {
+				console.log(data);
+			},
```
Comment (iccir919) [inline src/client/app/InviteeListEntry.jsx]: you already did this inviteelist. why not pass it down in state?

File: server/schema.js (hunk 2 above)
Comment (kay-lee) [inline server/schema.js]: That's nice way to save the day. Thanks.

File: server/routes/invitee_list.js (hunk 1 above)
Comment (kay-lee) [inline server/routes/invitee_list.js]: Yes. I didn't get to implement join table yet.

File: src/client/app/InviteeList.jsx (hunk 5 above)
Comment (kay-lee) [inline src/client/app/InviteeList.jsx]: nope. I didn't edit it yet.

File: src/client/styles.css (hunk 8 above)
Comment (kay-lee) [inline src/client/styles.css]: I appreciate your sophisticated taste in color.

**Comment by kay-lee** [comment]
- closing to submit completed feature in another branch

**Pull request closed without merging**